kartothek/io_components/metapartition.py

```diff
@@ -1715,3 +1712,5 @@ def _ensure_valid_indices(secondary_indices, mp_indices):
                 f"Expected index columns: {secondary_indices}"
                 f"Provided index: {mp_indices}"
             )
+    else:
+        raise ValueError("Please pass `secondary_indices` and `data` ")
```

Comment:
I do not get this case. This function should ensure that a explicitly passed index dictionary does provide all indices which are compatible with the existing dataset.
Not sure if hitting this clause is a real scenario. Either way, the exception message sounds misleading for this case

Reply:
Yes, you are right. I have made the checks for `data` and `secondary_indices` independent of each other. 

Thanks for pointing out :)